tools/src/main/scala/com/nvidia/spark/rapids/tool/profiling/AutoTuner.scala

```diff
@@ -0,0 +1,435 @@
+
+      if(numCores < MIN_PER_EXECUTOR_CORE_COUNT) {
+        comments :+= s"Number of cores per executor is very low. " +
+          s"It is recommended to have at least $MIN_PER_EXECUTOR_CORE_COUNT cores per executor."
+      }
+
+      if (systemProps.num_workers.nonEmpty) {
+        val numInstances = bestConfig.get_executor_instances
+        if (numCores * numInstances < systemProps.numCores) {
+          comments :+= "Not all cores in the machine are being used. " +
+            "It is recommended to use different machine."
+        }
+      }
+
+      bestConfig.set_executor_cores(numCores)
+
+      // Recommendation for 'spark.executor.memory' based on system memory, cluster scheduler
+      // and num of gpus
+      val sparkMaster = getSparkProperty(app, "spark.master")
+      val systemMemoryNum: Long = convertFromHumanReadableSize(systemProps.memory)
+      val extraSystemMemoryNum: Long = convertFromHumanReadableSize(MAX_EXTRA_SYSTEM_MEMORY)
+      val effectiveSystemMemoryNum: Long =
+        if (sparkMaster.contains("yarn") || sparkMaster.contains("k8s")) {
+          systemMemoryNum -  extraSystemMemoryNum -
+            convertFromHumanReadableSize(MIN_MEMORY_OVERHEAD)
+        } else systemMemoryNum - extraSystemMemoryNum
+      val maxExecutorMemNum: Long = convertFromHumanReadableSize(MAX_EXECUTOR_MEMORY)
+
+      val executorMemory: Long = if (systemProps.gpu_props != null) {
+        Math.min(effectiveSystemMemoryNum * 1.0 / systemProps.gpu_props.count,
+          maxExecutorMemNum).toLong
+      } else {
+        Math.min(effectiveSystemMemoryNum * 1.0 / numCores,
+          maxExecutorMemNum).toLong
+      }
+
+      if(executorMemory < convertFromHumanReadableSize(MIN_EXECUTOR_MEMORY)) {
+        comments :+= s"Executor memory is very low. " +
+          s"It is recommended to have at least $MIN_EXECUTOR_MEMORY"
+      }
+
+      bestConfig.set_executor_memory(convertToHumanReadableSize(executorMemory))
+
+      // Recommendation for 'spark.sql.shuffle.partitions' based on spill size
+      var shufflePartitions: Int = getSparkProperty(app, "spark.sql.shuffle.partitions")
+        .getOrElse("200").toInt
+
+      // TODO: Need to look at other metrics for GPU spills (DEBUG mode), and batch sizes metric
+      val totalSpilledMetrics = app.sqlTaskAggMetrics.map {
+        task => task.diskBytesSpilledSum + task.memoryBytesSpilledSum
+      }.sum
+      if (totalSpilledMetrics > 0) {
+        shufflePartitions *= DEFAULT_SHUFFLE_PARTITION_MULTIPLIER
+        // Could be memory instead of partitions
+        comments :+= "\"spark.sql.shuffle.partitions\" should be increased since spilling occurred."
+      }
+      bestConfig.set_shuffle_partitions(shufflePartitions)
+
+      // Recommendation for 'spark.sql.files.maxPartitionBytes' based on input size for each task
+      getSparkProperty(app, "spark.sql.files.maxPartitionBytes") match {
+        case None => bestConfig.set_max_partition_bytes(DEFAULT_MAX_PARTITION_BYTES)
```

Comment:
here if its not set we should still look at the stats to try to find best value, if we can't figure it out then default to this.  the default from spark is 128m

Reply:
Added calculation for this case.